modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java

```diff
@@ -100,6 +102,7 @@ public JavaSourceModelParser(ExtensionElement extensionElement,
 
     parseStructure();
     collectAdditionalModelProperties();
+    this.minMuleVersion = getMinMuleVersion().orElse(FIRST_MULE_VERSION);
```

Comment:
I don't understand this assignment. You're calling getMinMuleVersion() which is implemetned as:

```
@Override
  public Optional<MuleVersion> getMinMuleVersion() {
    return of(calculateSourceMinMuleVersion(sourceElement));
  }
  ```

Clients of the parser will call that getter, and the calculation will be performed everytime. Besides the efficiency concerns around that, why calculate this field if it isn't going to be returned ?

Reply:
You are right, fixing this.